packages/react-charting/UnitTests/AreaChartUT.test.tsx

```diff
@@ -0,0 +1,683 @@
+};
+
+runTest('_getOpacity', () => {
+  testWithoutWait(
+    'Should return fill-opacity of 0.7 if chart is not multi-stacked',
+    AreaChart,
+    { data: chartData },
+    container => {
+      const areas = getById(container, /graph-areaChart/i);
+      expect(areas.length).toBe(1);
+      expect(areas[0].getAttribute('fill-opacity')).toBe('0.7');
+    },
+  );
+
+  testWithoutWait(
+    'Should return fill-opacity of 0.7 if chart is multi-stacked',
+    AreaChart,
+    { data: chartDataMultiStacked },
+    container => {
+      const areas = getById(container, /graph-areaChart/i);
+      expect(areas.length).toBe(3);
+      expect(areas[0].getAttribute('fill-opacity')).toBe('0.7');
+    },
+  );
+
+  testWithoutWait(
+    'Should return fill-opacity of 0.7 if legend is highlighted and data is multi-stacked',
+    AreaChart,
+    { data: chartDataMultiStacked },
+    container => {
+      const legend = screen.queryByText('legend1');
+      expect(legend).toBeDefined();
+      fireEvent.mouseOver(legend!);
+      // Assert
+      const areas = getById(container, /graph-areaChart/i);
+      expect(areas[0].getAttribute('fill-opacity')).toEqual('0.7');
+    },
+  );
+
+  testWithoutWait(
+    'Should return fill-opacity of 0.1 if legend 1 is highlighted but legend 2 is not and data is multi-stacked',
+    AreaChart,
+    { data: chartDataMultiStacked },
+    container => {
+      const legend1 = screen.queryByText('legend1');
+      expect(legend1).toBeDefined();
+      fireEvent.mouseOver(legend1!);
+      // Assert
+      const areas = getById(container, /graph-areaChart/i);
+      expect(areas[1].getAttribute('fill-opacity')).toEqual('0.1');
+    },
+  );
+});
+
+runTest('_getLineOpacity', () => {
+  testWithoutWait('Should return 1 if chart is not multi-stacked', AreaChart, { data: chartData }, container => {
+    const areas = getById(container, /line-areaChart/i);
+    expect(areas.length).toBe(1);
+    expect(areas[0].getAttribute('opacity')).toBe('1');
+  });
+
+  testWithoutWait(
+    'Should return 0.3 if chart is multi-stacked and no legend is highlighted',
+    AreaChart,
+    { data: chartDataMultiStacked },
+    container => {
+      const areas = getById(container, /line-areaChart/i);
+      expect(areas.length).toBe(3);
+      expect(areas[0].getAttribute('opacity')).toBe('0.3');
+    },
+  );
+
+  testWithoutWait(
+    'Should return 1 if callout is visible and data is not multi-stacked',
+    AreaChart,
+    { data: chartData },
+    container => {
+      // Arrange
+      const areas = getById(container, /line-areaChart/i);
+      expect(areas.length).toBe(1);
+      fireEvent.mouseOver(areas[0]);
+      // Assert
+      expect(getById(container, /toolTipcallout/i)).toBeDefined();
+      expect(areas[0].getAttribute('opacity')).toBe('1');
+    },
+  );
+
+  testWithoutWait(
+    'Should return opacity of 0.1 if legend 1 is highlighted, legend 2 is not highlighted and data is multi-stacked',
```

Comment:
we dont have multi stacked in area chart. It is only stacked

Reply:
In this this._isMultiStackChart = stackedData && stackedData.length > 1 ? true : false; can the stackedData.length not be > 2 for area chart?

Reply:
I was talking about the naming convention. In practice it is a stacked chart. Not sure what multi stacked even mean here. But if the multistack term is being used in the code we can let it remain like this.